### PR TITLE
Refactor timer

### PR DESCRIPTION
**Pull request by spydon**
# Description

Refactor the timer class

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: doc/util.md [hunk 1]
```diff
@@ -73,15 +73,18 @@ class MyGame extends Game {
   @override
   void update(double dt) {
     countdown.update(dt);
-    if (countdown.isFinished()) {
-      // do something ...
+    if (countdown.finished) {
+      // Prefer the timer callback, but this is better in some cases
```
Comment (erickzanardo) [inline doc/util.md:77]: Suggestion: `Usually the timer callback could be used, but this could be used in some cases`

**Review by erickzanardo — COMMENTED** [review]
Shouldn't we cherry pick the changes that happened on this class on develop before? Since the class was renamed?

Other than that, LGTM already

**Review by luanpotter — APPROVED** [review]
LGTM

**Review by erickzanardo — APPROVED** [review]
LGTM

**Pull request merged**